### DISCUSSION TITLE
Project workflow methods.

### DISCUSSION THREAD

**Discussion by Philip-Clark** [club-40/odin-pantry#26] [General]
We are to a point where we will soon be writing code and actively developing this project, it is therefore of great importance that we decide how we will organize the development cycle. Sprints have been suggested as a good method, but any thoughts are welcome.

[Relevant dicord message](https://discord.com/channels/505093832157691914/1091794976045813770/1100231005534699522)

**Comment by taladan** [comment]
I don't have any solid preference of one way over another as this is my first time collaborating on a project this size as a software dev, instead of a different role.  I would ask a couple of clarifying points though:  

- If we go with a sprint methodology, or a hybridized sprint with looser time targets (say 1 month instead of 2 weeks) as per the DC discussion referenced, would that be a single sprint for both FE and BE, or would it be concurrent sprints where the FE team works on a feature and the BE team works on a separate (but possibly related) feature?  
- Who's going to determine what gets sprinted and in what order?  
- Who'll be in charge of each team and making sure that each goal is achievable within a sprint?
- What happens when/if a sprint goal isn't met?

There are probably more and better questions to be asked to hammer things down, but those are the first few that spring to mind as far as organizing things out.

**Reply by Philip-Clark** [reply]
~~These questions were addressed [here](https://discord.com/channels/505093832157691914/1091794976045813770/1100437935008006296)~~
Thanks for the awesome questions @taladan 🔥

**Reply by Ananth-Adhikarla** [reply]
1) if you cant finish all the tasks in the sprint it will spill over to the next sprint as backlog. I dont think we are using time here for anything so I dont think we need to look at 2 weeks or 1 month. 

2) FE and BE will work on the same sprint each with their own tasks but in some cases there could be a blocker where the data needs to come from the BE to FE so have to get the BE done first for example only then FE can progress.

3) maintainers / admin / volunteers will be creating tasks and putting them on the kanban board

4) after the tasks are put on the kanban board, you should have a choice to pick which task you want to work on pick one or multiple but if it doesnt work out then have to manually assign it based on skills I guess not sure yet

5) I dont think we have separate teams for frontend and backend but you can volunteer to be in charge if you want to one of the maintainers or admins will be by default then 

6) this is not a real sprint as you would see in a work place with deadlines to complete tasks since not everyone has the same amount of time and people could get stuck when working on something

reposted from discord
